gradio/blocks.py

```diff
@@ -1071,6 +1071,8 @@ async def call_function(
             else:
                 fn = block_fn.fn
 
+            fn = utils.get_function_with_locals(fn, self, event_id)
```

Comment:
You should call the output variable something else. The thing you return from this is not a coroutine so `iscoroutine` will always be false.

I can't get async functions to work (repro below) and I think it's because of that:

```python
import gradio as gr
import asyncio


async def warn_1():
    for i in range(10):
        gr.Warning(f"Warning {i}!")
        await asyncio.sleep(0.5)
    return "foo"

def warn_2():
    gr.Warning("Warning 2!")
    return "bar"

with gr.Blocks() as demo:
    t1 = gr.Textbox()
    t2 = gr.Textbox()

    button = gr.Button("Warning 1")
    button.click(warn_1, None, t1)

    button2 = gr.Button("Warning 2")
    button2.click(warn_2, None, t2)
    

demo.queue(concurrency_count=2).launch()
```

![image](https://github.com/gradio-app/gradio/assets/41651716/4c49f56d-49ae-4dab-8c25-cd47a383fe31)